nomad/structs/eval.go

```diff
@@ -0,0 +1,24 @@
+package structs
```

Comment:
As much as I love splitting this package up into more files, I'm not sure about just moving EvalDeleteRequest off on its own.

Should all Requests/Responses go together? Should we group by object type?

Do we even need a distinct type? I think we could just use the existing EvalDeleteRequest for everything and let `Allocs` be empty.

Not a big deal. Just a bit worried about splitting such similar types up might cause confusion.

Reply:
I have a preference that all RPC's have their own req and reply arguments. I find it makes reading easier, any changes in the future simpler as they only affect a single RPC, and avoids the requirement for understanding that certain fields are unused when using certain RPCs.

Arguably how definitions are grouped depends on what exists and factors such as import cycles. Nomad's structure involves a single pkg (structs) which contains all the RPC definitions, object structures, and helper functions. I therefore prefer to have individual files pertaining to each RPC endpoint object such as `Eval` or `ServiceRegistration`.

Reply:
I think we're in agreement that splitting up the giant `structs.go` is good. So I think there's just a follow-up refactoring PR to be done here after this gets merged, which is to move the _other_ eval-related stuff in `structs.go` into this file too.